Formula/pgvector.rb

```diff
@@ -0,0 +1,39 @@
+class Pgvector < Formula
+  desc "Open-source vector similarity search for Postgres"
+  homepage "https://github.com/pgvector/pgvector"
+  url "https://github.com/pgvector/pgvector/archive/v0.4.0.tar.gz"
+  sha256 "b76cf84ddad452cc880a6c8c661d137ddd8679c000a16332f4f03ecf6e10bcc8"
+  license "PostgreSQL"
+
+  depends_on "postgresql@14"
```

Comment:
Can we use `postgresql@15`? We probably do want to switch the other extensions as well.

Reply:
Just spent some time on this, but I'm not really sure the best way to do it. From what I can tell, the `postgesql@14` formula depends on being linked for both it and extensions to work.

- The `postgesql@14` formula depends on being linked since the [configure args](https://github.com/Homebrew/homebrew-core/blob/3fc145f53c2dc54a58e306d4f6f0de435e0a16ed/Formula/postgresql%4014.rb#L54-L56) use `HOMEBREW_PREFIX`
- Extensions depend on the `postgesql@14` formula being linked since they need to be placed inside the same directory as the built-in Postgres extensions

The easiest way would probably be to make `postgresql@14` keg-only and `postgresql@15` not (opposite of how they are now). I think it'd be good to change the `postgresql` alias to `postgresql@15` in that case, so `brew install postgresql` isn't keg-only. However, this still leaves the issue where the main Postgres formula must be linked to function (same as things are now).

Reply:
Could we merged this now for `@14` now, and do bumping all the extensions to `@15` in a separate issue?  I will even start another issue..

also TODO:
https://github.com/Homebrew/homebrew-core/blob/master/Formula/pg_cron.rb
https://github.com/Homebrew/homebrew-core/blob/master/Formula/pg_partman.rb
https://github.com/Homebrew/homebrew-core/blob/master/Formula/citus.rb
https://github.com/Homebrew/homebrew-core/blob/master/Formula/postgis.rb
https://github.com/Homebrew/homebrew-core/blob/master/Formula/pgroonga.rb
https://github.com/Homebrew/homebrew-core/blob/master/Formula/pgrouting.rb
https://github.com/Homebrew/homebrew-core/blob/master/Formula/temporal_tables.rb

Reply:
https://github.com/Homebrew/homebrew-core/pull/125631

Reply:
@carlocab what do you think about keeping this at `@14` until there's a clean way to move all the extensions together to `@15`?

Reply:
Fine with me.